src/features/tokens/metadata.ts

```diff
@@ -1,25 +1,48 @@
+import { ethers } from 'ethers';
+
+import { TokenType } from '@hyperlane-xyz/hyperlane-token';
+
 import { tokenList } from '../../consts/tokens';
-import { areAddressesEqual } from '../../utils/addresses';
 import { logger } from '../../utils/logger';
 
-import { TokenListSchema, TokenMetadata } from './types';
+import { TokenMetadata, WarpTokenConfig, WarpTokenConfigSchema } from './types';
 
 let tokens: TokenMetadata[];
 
 export function getAllTokens() {
   if (!tokens) {
-    const result = TokenListSchema.safeParse(tokenList);
-    if (!result.success) {
-      logger.error('Invalid token config', result.error);
-      throw new Error(`Invalid token config: ${result.error.toString()}`);
-    }
-    tokens = result.data;
+    tokens = parseTokenConfigs(tokenList);
   }
   return tokens;
 }
 
-export function getTokenMetadata(chainId: number, tokenAddress: Address) {
-  return getAllTokens().find(
-    (t) => t.chainId == chainId && areAddressesEqual(t.address, tokenAddress),
-  );
+function parseTokenConfigs(configList: WarpTokenConfig): TokenMetadata[] {
+  const result = WarpTokenConfigSchema.safeParse(configList);
+  if (!result.success) {
+    logger.error('Invalid token config', result.error);
+    throw new Error(`Invalid token config: ${result.error.toString()}`);
+  }
+  const parsedConfig = result.data;
+  const tokenMetadata: TokenMetadata[] = [];
+  for (const token of parsedConfig) {
+    const { type, chainId, name, symbol, decimals, logoURI } = token;
+    const commonFields = { chainId, name, symbol, decimals, logoURI };
+    if (type == TokenType.collateral) {
+      tokenMetadata.push({
+        ...commonFields,
+        type: TokenType.collateral,
+        hypWrapperAddress: token.hypCollateralAddress,
+        address: token.address,
+      });
+    } else if (type == TokenType.native) {
+      tokenMetadata.push({
+        ...commonFields,
+        type: TokenType.native,
+        hypWrapperAddress: token.hypNativeAddress,
+        // Note, using 0x000... address to help identify native tokens
```

Comment:
Why is this necessary given `type`?

Reply:
It's not strictly necessary but I found that for other parts of the code (mostly UI code) it was helpful for *all* metadata objects to have an addresss